cnd/src/config.rs

```diff
@@ -61,14 +61,14 @@ impl From<Bitcoin> for file::Bitcoin {
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
 pub struct Ethereum {
     pub chain_id: ethereum::ChainId,
-    pub parity: Parity,
+    pub web3_url: Url,
```

Comment:
That is kinda inconsistent with the bitcoin side of things now.

Why not leave the section there but name it `web3` and the field can remain `node_url`?

Reply:
For Bitcoin we do specify that we support `bitcoind`. Hence, it is sufficient to have `node_url` terminology as the context tells us already that the given node must be `bitcoind`.

However, in the case of Ethereum, we support a generic node and only named the section `ethereum`.
By naming the url `web3_url` we do specify that we are expecting the `web3` api.

Reply:
What I was pointing out is that we are missing a layer now for Ethereum. From what you are saying, we would have to change the `node_url` for Bitcoin to `bitcoind_node_url` to be consistent again.

The point of the additional "layer" in the configuration was to make it trivial to extend with other connectors even if we just support one at the moment.

I don't mind removing the "connector" layer but it should be consistent for all blockchains we support. At the moment we have:

```
[bitcoin]
network = "regtest"

[bitcoin.bitcoind]
node_url = "http://localhost:18443/"

[ethereum]
chain_id = 1337

[ethereum.parity]
node_url = "http://localhost:8545/"
```

but with this change, we have:

```
[bitcoin]
network = "regtest"

[bitcoin.bitcoind]
node_url = "http://localhost:18443/"

[ethereum]
chain_id = 1337
web3_url = "http://localhost:8545/"
```

Reply:
I'd like to see, what ever way is chosen, that its uniform across ledgers also please.

Reply:
Thanks for clarifying. I don't think it makes to align with Bitcoin just for the sake of uniformity.

In this case, I would suggest:

```
[bitcoin]
network = "regtest"

[bitcoin.bitcoind]
node_url = "http://localhost:18443/"

[ethereum]
chain_id = 1337

[ethereum.web3]
url = "http://localhost:8545/"
```

Note that a web3 endpoint is not necessarily a `node`. Infura is probably a cloud base service and most likely request does not target a specific `node`. In any case, what matters is that we expect a web3 api and hence the `node` terminology is unnecessary at best.

I am happy to also change `bitcoin.bitcoind.node_url` to `url` for the sake of _uniformity_. But again, I do not see any point in doing so.

Reply:
The reason why we had `parity` in the first place was that we explicitly said we only support Parity as we never tested against Geth. Now it's the other way around: we will keep testing against Geth but don't know about Parity for sure. 

I'm not proposing a solution but just wanted to mention this.

Reply:
> The reason why we had `parity` in the first place was that we explicitly said we only support Parity as we never tested against Geth. Now it's the other way around: we will keep testing against Geth but don't know about Parity for sure.
> 
> I'm not proposing a solution but just wanted to mention this.

Thanks for the input. Is that something we still want to uphold? If someone come to us and they use infura and we need to do a code change to support infura, would we reject and say "use geth" or we will attempt to support any reasonable Ethereum software?

Reply:
To avoid ambiguity, what about just changing `parity` to `geth` and leaving the rest as is?

Infura uses API-keys to my knowledge, so just providing a URL won't work for that even though it is a web3 compatible API. (see here: https://blog.infura.io/getting-started-with-infura-28e41844cc89/).

If we say `ethereum.geth.node_url`, it is unambigous what we officially support and the extension path for more connectors is trivial as-well.

Maybe `ethereum.geth.web3_url` is even better to make it clear that we don't want a websocket endpoint or anything like that!

If over time, we find a more abstract model for representing the same configuration, we can always refactor towards that :)

Reply:
> Thanks for the input. Is that something we still want to uphold? 

Yes, I think it would make sense to explicitly state what we support, e.g. Geth instead of Parity. 

> If someone come to us and they use infura and we need to do a code change to support infura, would we reject and say "use geth" or we will attempt to support any reasonable web3 provider?

Good question. I don't think there is an answer for this right now and I would wait until this happens and decide then.

Reply:
> Yes, I think it would make sense to explicitly state what we support, e.g. Geth instead of Parity.

Ok, I did not think this would be the answer. I'll proceed with the change.